internal/metrics/cmd.go

```diff
@@ -40,45 +40,56 @@ func (m Metrics) getCPU() cpuInfo {
 
 	r := runCmd(m.cpuInfoCmd)
 
-	for result := range filter(r, `{"field": *"(.*)", *"data": *"(.*)"},`, true) {
-		if result.err != nil {
-			log.Infof("Couldn't get CPU info: "+utils.ErrFormat, result.err)
-			return cpuInfo{}
-		}
+	result, err := parseJSON(r, &Lscpu{})
 
-		key, v := result.r[0], result.r[1]
-
-		switch strings.TrimSpace(key) {
-		case "CPU op-mode(s):":
-			c.OpMode = v
-		case "CPU(s):":
-			c.CPUs = v
-		case "Thread(s) per core:":
-			c.Threads = v
-		case "Core(s) per socket:":
-			c.Cores = v
-		case "Socket(s):":
-			c.Sockets = v
-		case "Vendor ID:":
-			c.Vendor = v
-		case "CPU family:":
-			c.Family = v
-		case "Model:":
-			c.Model = v
-		case "Stepping:":
-			c.Stepping = v
-		case "Model name:":
-			c.Name = v
-		case "Virtualization:":
-			c.Virtualization = v
-		case "Hypervisor vendor:":
-			c.Hypervisor = v
-		case "Virtualization type:":
-			c.VirtualizationType = v
+	if err != nil {
+		log.Infof("Couldn't get CPU info: "+utils.ErrFormat, err)
+		return cpuInfo{}
+	}
+
+	lscpu := result.(*Lscpu)
```

Comment:
Here, if the casting for `$reasons` fails, the program will panic, which is what we always try to avoid in Go. It would be better to assign the success check when casting with `, ok` and check for the `ok` value to return an error.
Typically:

```
lscpu, ok := result.(*Lscpu)
if !ok {
  log.Infof("Couldn't get CPU info, could not convert to a valid Lscpu entry: %v", result)
}

Reply:
Done!